python/cmsis_svd/parser.py

```diff
@@ -363,375 +353,486 @@ def for_mcu(cls, package_root, mcu):
-                size=size,
-                access=access,
-                protection=protection,
-                reset_value=reset_value,
-                reset_mask=reset_mask,
-                display_name=display_name,
-                alternate_group=alternate_group,
-                modified_write_values=modified_write_values,
-                read_action=read_action,
-            )
+        if field.dim is not None:
+            return SVDFieldArray(meta_field=field)
         else:
-            # the node represents a register array
-            if dim_index_text is None:
-                dim_indices = list(range(0, dim))  # some files omit dimIndex
-            elif ',' in dim_index_text:
-                dim_indices = dim_index_text.split(',')
-            elif '-' in dim_index_text:  # some files use <dimIndex>0-3</dimIndex> as an inclusive inclusive range
-                start, stop = dim_index_text.split('-')
-
-                if start.isalpha() and stop.isalpha():
-                    start_val = ord(start)
-                    stop_val = ord(stop)
-
-                    dim_indices = [
-                        chr(val)
-                        for val in range(start_val, stop_val + 1)
-                    ]
-
-                elif start.isdigit() and stop.isdigit():
-                    start_val = int(start)
-                    stop_val = int(stop)
-
-                    dim_indices = [
-                        str(val)
-                        for val in range(start_val, stop_val + 1)
-                    ]
+            return field
 
-                else:
-                    raise NotImplementedError(
-                        f'DimIndex={dim_index_text} is not supported.'
-                    )
-            else:
-                raise ValueError("Unexpected dim_index_text: %r" % dim_index_text)
-
-            # yield `SVDRegisterArray` (caller will differentiate on type)
-            return SVDRegisterArray(
-                name=name,
-                fields=fields,
-                derived_from=derived_from,
-                description=description,
-                address_offset=address_offset,
-                size=size,
-                access=access,
-                protection=protection,
-                reset_value=reset_value,
-                reset_mask=reset_mask,
-                display_name=display_name,
-                alternate_group=alternate_group,
-                modified_write_values=modified_write_values,
-                read_action=read_action,
-                dim=dim,
-                dim_indices=dim_indices,
-                dim_increment=dim_increment,
-            )
+    def _parse_register(
+        self, register_node: Element
+    ) -> Union[SVDRegister, SVDRegisterArray]:
+        fields = [self._parse_field(f) for f in register_node.findall('./fields/field')
+                  if not self.remove_reserved or 'reserved' not in f.name.lower()]
```

Comment:
May accidentally filter attributes called "preserved*" …

Maybe have this as a function `isReservedAttribute(name, ctx)` that can easily be extended if such conflicts are found in the wild (`ctx` being some context of the field in question).

Reply:
Fixed.